app/src/app/.well-known/oauth-protected-resource/route.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @swagger
+ * /.well-known/oauth-protected-resource:
+ *   get:
+ *     tags:
+ *       - oauth
+ *     operationId: getOauthProtectedResource
+ *     summary: OAuth Protected Resource Discovery (RFC 8414)
+ *     description: |
+ *       Advertises OAuth 2.0 protected resources and their endpoints.
+ *       This is part of the MCP server discovery standard.
+ *     responses:
+ *       200:
+ *         description: Protected resource metadata
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 resource:
+ *                   type: string
+ *                   description: The base URI of the protected resource
+ *                 authorization_servers:
+ *                   type: array
+ *                   items:
+ *                     type: string
+ *                   description: Array of authorization server issuer identifiers
+ *                 scopes_supported:
+ *                   type: array
+ *                   items:
+ *                     type: string
+ *                   description: OAuth scopes supported by this resource server
+ *                 bearer_methods_supported:
+ *                   type: array
+ *                   items:
+ *                     type: string
+ *                   description: Bearer token usage methods supported
+ *                 resource_documentation:
+ *                   type: string
+ *                   description: URL pointing to documentation about the resource
+ */
+import { NextResponse } from "next/server";
+import { logger } from "@/services/logger";
+
+export const GET = async (req: Request) => {
+  const url = new URL(req.url);
+  const origin = process.env.HOST || url.origin;
+  
+  logger.debug({ origin }, "OAuth protected resource discovery requested");
+
+  return NextResponse.json({
+    resource: origin,
+    authorization_servers: [origin],
+    scopes_supported: ["read", "write"],
+    bearer_methods_supported: ["header"],
+    resource_documentation: "https://github.com/Open-Earth-Foundation/CityCatalyst/wiki/CityCatalyst-Backend-API",
```

Comment:
This isn't quite up do date, might make sense to point to the Swagger docs page instead? Using the current HOST value (`origin` here) and appending `/docs`.